subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -453,40 +837,104 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
 	p.remainder = remainder;
 	p.lazy = lazy;
 	p.force = force;
-	p.param = lll;
-	mfy.param = &p;
+	p.param = &adv_iso->lll;
+	mfy_lll_prepare.param = &p;
 
 	/* Kick LLL prepare */
-	ret = mayfly_enqueue(TICKER_USER_ID_ULL_HIGH,
-			     TICKER_USER_ID_LLL, 0, &mfy);
+	ret = mayfly_enqueue(TICKER_USER_ID_ULL_HIGH, TICKER_USER_ID_LLL, 0,
+			     &mfy_lll_prepare);
 	LL_ASSERT(!ret);
 
 	DEBUG_RADIO_PREPARE_A(1);
-#endif
 }
 
-static void tx_lll_flush(void *param)
+static void ticker_op_cb(uint32_t status, void *param)
 {
-	/* TODO: LLL support for ADV ISO */
-	/* TODO: Send terminate complete event to host */
-#if 0
-	/* TODO: Flush TX */
-	struct ll_adv_iso *lll = param;
-#endif
+	*((uint32_t volatile *)param) = status;
 }
 
-static void ticker_op_stop_cb(uint32_t status, void *param)
+static void ticker_stop_op_cb(uint32_t status, void *param)
 {
-	uint32_t retval;
 	static memq_link_t link;
-	static struct mayfly mfy = {0, 0, &link, NULL, tx_lll_flush};
+	static struct mayfly mfy = {0, 0, &link, NULL, adv_iso_disable};
+	uint32_t ret;
 
 	LL_ASSERT(status == TICKER_STATUS_SUCCESS);
 
+	/* Check if any pending LLL events that need to be aborted */
+	mfy.param = param;
+	ret = mayfly_enqueue(TICKER_USER_ID_ULL_LOW,
+			     TICKER_USER_ID_ULL_HIGH, 0, &mfy);
+	LL_ASSERT(!ret);
+}
+
+static void adv_iso_disable(void *param)
+{
+	struct ll_adv_iso_set *adv_iso;
+	struct ull_hdr *hdr;
+
+	/* Check ref count to determine if any pending LLL events in pipeline */
+	adv_iso = param;
+	hdr = &adv_iso->ull;
+	if (ull_ref_get(hdr)) {
+		static memq_link_t link;
+		static struct mayfly mfy = {0, 0, &link, NULL, lll_disable};
+		uint32_t ret;
+
+		mfy.param = &adv_iso->lll;
+
+		/* Setup disabled callback to be called when ref count
+		 * returns to zero.
+		 */
+		LL_ASSERT(!hdr->disabled_cb);
+		hdr->disabled_param = mfy.param;
+		hdr->disabled_cb = disabled_cb;
+
+		/* Trigger LLL disable */
+		ret = mayfly_enqueue(TICKER_USER_ID_ULL_HIGH,
+				     TICKER_USER_ID_LLL, 0, &mfy);
+		LL_ASSERT(!ret);
+	} else {
+		/* No pending LLL events */
+		disabled_cb(&adv_iso->lll);
+	}
+}
+
+static void disabled_cb(void *param)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, tx_lll_flush};
+	uint32_t ret;
```

Comment:
This Christmas tree looks a bit odd ;)

Reply:
As there is static initialization, this one is an exception.

That said, I am getting the feeling that controller is exploiting use of mayfly(s) using static allocations, there can be RAM optimization by carefully using dynamic allocations, this is something for future enhancement once we have some maturity in ISO implementation in controller.